server/protocols/webadmin/webadmin-mailbox/src/test/java/org/apache/james/webadmin/routes/UserMailboxesRoutesTest.java

```diff
@@ -1284,6 +1346,75 @@ void userReprocessingShouldReturnTaskDetailsWhenFailingAtTheMailboxLevel() throw
+                    .getId();
+                ComposedMessageId result2 = mailboxManager.getMailbox(INBOX, systemSession)
+                    .appendMessage(
+                        MessageManager.AppendCommand.builder().build("header: value\r\n\r\nbody"),
+                        systemSession)
+                    .getId();
+
+                mailboxManager.getMailbox(INBOX, systemSession)
+                    .appendMessage(
+                        MessageManager.AppendCommand.builder().build("header: value\r\n\r\nbody"),
+                        systemSession);
+                mailboxManager.getMailbox(INBOX, systemSession)
+                    .appendMessage(
+                        MessageManager.AppendCommand.builder().build("header: value\r\n\r\nbody"),
+                        systemSession);
+
+                List<MessageResult> messages = messageIdManager.getMessages(ImmutableList.of(result1.getMessageId(), result2.getMessageId()), FetchGroup.MINIMAL, systemSession);
+
+                Flags newFlags1 = new Flags(Flags.Flag.DRAFT);
+                UpdatedFlags updatedFlags1 = UpdatedFlags.builder()
+                    .uid(result1.getUid())
+                    .modSeq(messages.get(0).getModSeq())
+                    .oldFlags(new Flags())
+                    .newFlags(newFlags1)
+                    .build();
+
+                Flags newFlags2 = new Flags(Flags.Flag.DELETED);
+                UpdatedFlags updatedFlags2 = UpdatedFlags.builder()
+                    .uid(result2.getUid())
+                    .modSeq(messages.get(1).getModSeq())
+                    .oldFlags(new Flags())
+                    .newFlags(newFlags2)
+                    .build();
+
+                // We update on the searchIndex level to try to create inconsistencies
+                searchIndex.update(systemSession, mailbox, ImmutableList.of(updatedFlags1, updatedFlags2)).block();
+
+                String taskId = with()
+                    .queryParam("task", "reIndex")
+                    .queryParam("mode", "correct")
+                    .post()
+                    .jsonPath()
+                    .get("taskId");
+
+                given()
+                    .basePath(TasksRoutes.BASE)
+                .when()
+                    .get(taskId + "/await")
+                .then()
+                    .body("status", Matchers.is("completed"))
+                    .body("taskId", Matchers.is(notNullValue()))
+                    .body("type", Matchers.is(UserReindexingTask.USER_RE_INDEXING.asString()))
+                    .body("additionalInformation.successfullyReprocessedMailCount", Matchers.is(2))
+                    .body("additionalInformation.failedReprocessedMailCount", Matchers.is(0))
+                    .body("startedDate", Matchers.is(notNullValue()))
+                    .body("submitDate", Matchers.is(notNullValue()))
+                    .body("completedDate", Matchers.is(notNullValue()));
+            }
```

Comment:
Do additionalInformation.successfullyReprocessedMailCount includes mails that did not had inconsistencies (I expect it should)

Reply:
I'm missing the following test:

 - Verify in ElasticSearch that an inconsistency is fixed after running the task
 - Verify in ElasticSearch that nothing changes when there is no inconsistency after running the task.
 - Verify that "orphan" messages in ElasticSearch are well deleted. (or at least document the limitation)

Reply:
> Do additionalInformation.successfullyReprocessedMailCount includes mails that did not had inconsistencies (I expect it should)

Mails that did not have inconsistencies are not being reprocessed, so I would on the contrary expect that it should not be included in the count of mails reprocessed?

Reply:
> Verify that "orphan" messages in ElasticSearch are well deleted. (or at least document the limitation) 

The limitation has been documented: https://github.com/linagora/james-project/pull/3429/commits/823ad41cdc0bb97d4c6028bacb558152fdf4bc7b

Reply:
> Mails that did not have inconsistencies are not being reprocessed, so I would on the contrary expect that it should not be included in the count of mails reprocessed?

I would rather expect to differenciate `processedMails` from `fixedInconsistencies`, no?

Reply:
>  The limitation has been documented: 823ad41

A @Ignored test is the best documentation you could ever write for this ;-)

Reply:
Should we apply the throttling on all emails? Or only on the ones having inconsistencies?

In other tasks we applied that for all emails.